compiler/rustc_codegen_gcc/example/alloc_example.rs

```diff
@@ -1,48 +0,0 @@
-#![feature(start, core_intrinsics, alloc_error_handler, lang_items)]
```

Comment:
I would actually prefer to keep this test, if it's not too much trouble.
During the sync I'm trying to do with the rust repo, I just got errors related to alloc, so I could still use this test.
Sorry for the change of mind.
The rest of the changes in cg_gcc looks good to me.
Thanks!

Reply:
Do you know which CI runner runs the GCC tests, so that we can do a try build?

Reply:
I do not know for sure (@GuillaumeGomez would know), but if I understand [this line](https://github.com/rust-lang/rust/blob/master/src/ci/github-actions/jobs.yml#L117) correctly, it would be the job `x86_64-gnu-llvm-18`.

Reply:
Two docker images (`x86_64-gnu-llvm-18` and `x86_64-gnu-llvm-19`) run gcc backend tests. You can find them by grepping on `build-gccjit.sh`.

Reply:
Okay it runs in PR CI then, no needs for a try job for that.